### PR TITLE
feat: upgrading rtsf-at-checkout-device-scale service to EdgeX2.2 phase 1

### PR DESCRIPTION
**Pull request by vli11**
Signed-off-by: Valina Li <valina.li@intel.com>

### CODE REVIEW

**Review by marcpfuller — APPROVED** [review]
LGTM

File: rtsf-at-checkout-device-scale/Dockerfile [hunk 1]
```diff
@@ -1,7 +1,8 @@
-# Copyright © 2020 Intel Corporation. All rights reserved.
+# Copyright © 2022 Intel Corporation. All rights reserved.
 # SPDX-License-Identifier: BSD-3-Clause
 
-FROM golang:1.13-alpine AS builder
+# FROM golang:1.13-alpine AS builder
```
Comment (jim-wang-intel) [inline rtsf-at-checkout-device-scale/Dockerfile:4]: remove dead code

File: rtsf-at-checkout-device-scale/driver/scaleDriver.go [hunk 2]
```diff
@@ -90,10 +91,17 @@ func processScaleData(scaleData map[string]interface{}, deviceResName string) (*
 		return nil, err
 	}
 
-	now := time.Now().UnixNano() / int64(time.Millisecond)
+	commandvalue, err := dsModels.NewCommandValueWithOrigin(
```
Comment (jim-wang-intel) [inline rtsf-at-checkout-device-scale/driver/scaleDriver.go:94]: commandvalue --> commandValue camel cases as usual practice

**Review by brian-intel — APPROVED** [review]
LGTM after addressing Jim's comment

**Review by jim-wang-intel — APPROVED** [review]

File: rtsf-at-checkout-device-scale/Dockerfile [hunk 3]
```diff
@@ -12,21 +12,22 @@ LABEL license='SPDX-License-Identifier: Apache-2.0' \
 RUN sed -e 's/dl-cdn[.]alpinelinux.org/nl.alpinelinux.org/g' -i~ /etc/apk/repositories
 
 # add git for go modules
-RUN apk update && apk add make git
+RUN apk update && apk add --no-cache make git gcc libc-dev zeromq-dev linux-headers
```
Comment (lenny-intel) [inline rtsf-at-checkout-device-scale/Dockerfile:15]: We can avoid `gcc libc-dev zeromq-dev linux-headers`  if we add the `no_zmq` tag to the build.

File: rtsf-at-checkout-device-scale/Dockerfile [hunk 4]
```diff
@@ -12,21 +12,22 @@ LABEL license='SPDX-License-Identifier: Apache-2.0' \
...
 
 COPY go.mod .
 
 RUN go mod download
 
 COPY . .
 
-RUN CGO_ENABLED=0 go build -o device-scale cmd/main.go
+RUN go mod tidy
+RUN CGO_ENABLED=1 GOOS=linux go build -o device-scale -ldflags='-s -w' -a -installsuffix cgo cmd/main.go
```
Comment (lenny-intel) [inline rtsf-at-checkout-device-scale/Dockerfile:24]: Why adding `ldflags='-s -w'` and `GOOS=linux` ? 
I should be able to build and run the binary on Windows

```suggestion
go build -tags no_zmq -o device-scale cmd/main.go
```

**Review by lenny-intel — CHANGES_REQUESTED** [review]

**Review by lenny-intel — COMMENTED** [review]
Configuration, Profile and Device definition still need to be migrated.

File: rtsf-at-checkout-device-scale/Dockerfile [hunk 5]
```diff
@@ -12,21 +12,22 @@ LABEL license='SPDX-License-Identifier: Apache-2.0' \
...
 
 # Next image - Copy built Go binary into new workspace
-FROM golang:1.13-alpine
+FROM golang:1.16-alpine
```
Comment (jim-wang-intel) [inline rtsf-at-checkout-device-scale/Dockerfile:27]:
```suggestion
FROM golang:1.18-alpine
```

**Review by lenny-intel — APPROVED** [review]
LGTM

**Pull request merged**